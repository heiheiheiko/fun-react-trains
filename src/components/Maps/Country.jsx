import React from "react";
import PropTypes from 'prop-types';
import { Geography } from "react-simple-maps";
import { colorDarkest, colorDarker } from "../../config/colors";

const mapStyles = {
  fill: colorDarkest,
  stroke: colorDarker,
  strokeWidth: 0.25,
  outline: "none",
}

function CountryMap(props) {
  return (
    <Geography
      geography={props.geography}
      projection={props.projection}
      style={{
        default: mapStyles,
        hover: mapStyles,
        pressed: mapStyles,
      }}
    />
  )
}

CountryMap.propTypes = {
  geography: PropTypes.object,
  projection: PropTypes.func,
};

export default CountryMap;