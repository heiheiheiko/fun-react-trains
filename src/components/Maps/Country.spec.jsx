import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer'
import SpecContainer from '../../../__specs__/helpers/SpecIntelContainer';

import Country from './Country';

jest.mock('react-simple-maps', () => ({
  Geography: "Geography"
}))

describe('<Country>', () => {
  it('could be renderd', () => {
    const mountedSpecContainer = mount(<SpecContainer><Country/></SpecContainer>);
    expect(mountedSpecContainer.find(Country).length).toEqual(1);
  });

  it('matched the snapshot', () => {
    const specContainer = <SpecContainer><Country/></SpecContainer>;
    const renderedValue =  renderer.create(specContainer).toJSON()
    expect(renderedValue).toMatchSnapshot();
  });
});