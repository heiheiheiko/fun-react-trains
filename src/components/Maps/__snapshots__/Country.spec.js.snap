// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Country> matched the snapshot 1`] = `
<div>
   
  <Geography
    style={
      Object {
        "default": Object {
          "fill": "#2c3039",
          "outline": "none",
          "stroke": "#3b3e48",
          "strokeWidth": 0.25,
        },
        "hover": Object {
          "fill": "#2c3039",
          "outline": "none",
          "stroke": "#3b3e48",
          "strokeWidth": 0.25,
        },
        "pressed": Object {
          "fill": "#2c3039",
          "outline": "none",
          "stroke": "#3b3e48",
          "strokeWidth": 0.25,
        },
      }
    }
  />
   
</div>
`;
